### PR TITLE
CCM-4893: update email to show subject line

### PR DESCRIPTION
**Pull request by alibombaye**
## Description
Updates the the email sent to the user.

NOTE: The attachment has been removed also since it's no longer required.

When an email template is submitted. The subject line is added
Screenshot 2024-10-08 at 11 37 08

When a non-email template is submitted. The subject line is not added
Screenshot 2024-10-08 at 11 38 17

## Context
Emails the user with the details added.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: amplify/__tests__/functions/send-email/handler.test.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ type HandlerCallbackType = Parameters<
 beforeAll(() => {
   jest.useFakeTimers();
   jest.setSystemTime(new Date('2022-01-01 10:00'));
-  process.env.NOTIFY_DOMAIN_NAME = 'test.notify.nhs.uk'
+  process.env.NOTIFY_DOMAIN_NAME = 'test.notify.nhs.uk';
```
Comment (chris-elliott-nhsd) [inline amplify/__tests__/functions/send-email/handler.test.ts:21]: i think it's good practice to unset test environment variables at the end of tests when you don't need them anymore - it just ensures they don't mess with other tests (may not affect anything now but could cause hard to diagnose issues in future)

File: amplify/functions/send-email/email-template.ts [hunk 2]
```diff
@@ -8,12 +8,14 @@ const encoder = (rawText: string) => encode(rawText).replaceAll('\n', '<br />');
 export const emailTemplate = (
   templateId: string,
   templateName: string,
-  templateMessage: string
+  templateMessage: string,
+  templateSubjectLine?: string | null
```
Comment (chris-elliott-nhsd) [inline amplify/functions/send-email/email-template.ts:12]: this is allowing for separate cases for when it's passed in as null and when it's not passed in - is this needed?

File: amplify/__tests__/functions/send-email/handler.test.ts (hunk 1 above)
Comment (alibombaye) [reply amplify/__tests__/functions/send-email/handler.test.ts:23]: Yeah that's a fair shout. I've reset the envVar back to what it was in an afterAll if that's what you meant by the unset.

File: amplify/functions/send-email/email-template.ts (hunk 2 above)
Comment (alibombaye) [reply amplify/functions/send-email/email-template.ts:12]: The issue is we have is when we save the template and as a result get the returned template from amplify. The subjectLine for non email templates this value will be null.

But it probably doesn't make sense for it to ever be optional in this instance. Will update

**Review by chris-elliott-nhsd — APPROVED** [review]

File: src/utils/form-actions.ts [hunk 3]
```diff
@@ -130,13 +130,15 @@ export async function getTemplate(
 export async function sendEmail(
   templateId: string,
   templateName: string,
-  templateMessage: string
+  templateMessage: string,
+  templateSubjectLine: string | null
 ) {
   const res = await getAmplifyBackendClient().queries.sendEmail({
-    recipientEmail: 'england.test.cm@nhs.net',
+    recipientEmail: 'fawad.ali3@nhs.net',
```
Comment (mark-r-bjss) [inline src/utils/form-actions.ts:137]: Nope

**Review by mark-r-bjss — DISMISSED** [review]

**Review by bhansell1 — APPROVED** [review]

**Pull request merged**